cmd_wait_ready.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/signal"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	connectionRetryInterval = time.Millisecond * 250
+)
+
+type waitReadyCommand struct {
+	RPCServer string        `long:"rpcserver" description:"The host:port of lnd's RPC listener"`
+	Timeout   time.Duration `long:"timeout" description:"The maximum time we'll wait for lnd to become ready; 0 means wait forever"`
+}
+
+func newWaitReadyCommand() *waitReadyCommand {
+	return &waitReadyCommand{
+		RPCServer: defaultRPCServer,
+	}
+}
+
+func (x *waitReadyCommand) Register(parser *flags.Parser) error {
+	_, err := parser.AddCommand(
+		"wait-ready",
+		"Wait for lnd to be fully ready",
+		"Wait for lnd to be fully started, unlocked and ready to "+
+			"serve RPC requests; will wait and block forever "+
+			"until either lnd reports its status as ready or the "+
+			"given timeout is reached; the RPC connection to lnd "+
+			"is re-tried indefinitely and errors are ignored (or "+
+			"logged in verbose mode) until success or timeout; "+
+			"requires lnd v0.13.0-beta or later to work",
```

Comment:
The readme should probably have a Requirements section that mentions the LND versions supported by this utility. Hopefully anyone who adds this is running a recent version, but might as well specify it to be safe. That docs section would also provide a place to add requirements for optional components, like kubernetes version, etc.